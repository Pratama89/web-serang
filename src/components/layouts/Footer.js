export default function Footer() {
  // Mendapatkan tahun saat ini
  const currentYear = new Date().getFullYear();

  return (
    <div className="flex md:flex-row xs:flex-col justify-center bg-primary text-slate-300 text-sm xs:text-sm p-6 text-center dark:bg-neutral-700">
      <span>Â© {currentYear} Copyright: CV. Mutiara Elasticindo</span>
      {/* <span>Website Designed by : MasCipto</span> */}
    </div>
  );
}
