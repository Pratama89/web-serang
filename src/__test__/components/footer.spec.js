import Footer from "@/components/layouts/Footer";
import { render, screen } from "@testing-library/react";
import { describe } from "node:test";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/footer",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}));

describe("Footer", () => {
  it("should render", () => {
    const page = render(<Footer />);
    // expect(screen.getByTestId("title").textContent).toBe("Header");
    expect(page).toMatchSnapshot();
  });
});
