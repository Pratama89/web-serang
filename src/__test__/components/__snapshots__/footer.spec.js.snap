// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex md:flex-row xs:flex-col justify-between bg-primary text-slate-300 text-sm xs:text-xs p-6 text-center dark:bg-neutral-700"
      >
        <span>
          © 2023 Copyright: Mutiara Laminating
        </span>
        <span>
          Website Designed by : mccreation
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex md:flex-row xs:flex-col justify-between bg-primary text-slate-300 text-sm xs:text-xs p-6 text-center dark:bg-neutral-700"
    >
      <span>
        © 2023 Copyright: Mutiara Laminating
      </span>
      <span>
        Website Designed by : mccreation
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
